bin/varnishd/cache/cache_obj.c

```diff
@@ -253,13 +271,18 @@ ObjWaitExtend(const struct worker *wrk, const struct objcore *oc, uint64_t l)
 	CHECK_OBJ_NOTNULL(oc->boc, BOC_MAGIC);
 	Lck_Lock(&oc->boc->mtx);
 	while (1) {
-		rv = oc->boc->len_so_far;
+		rv = oc->boc->fetched_so_far;
 		assert(l <= rv || oc->boc->state == BOS_FAILED);
+		if (oc->flags & OC_F_PRIVATE) {
```

Comment:
Same as HFM/HFP comment above.

Reply:
Same comment as above, test `oc->boc->pass_prefetch > 0` to signal changes in `oc->boc->delivered_so_far`.